src/Window.cpp

```diff
@@ -1279,15 +1311,65 @@ void Window::updateMouseProgress(double progress) {
   double fraction = 1.0 - (remaining / required);
 
   // Calculate the current translation and rotation
-  Coordinate startingTranslation = getCoordinate(m_startingPosition);
+  Coordinate startingTranslation    = getCoordinate(m_startingPosition);
   Coordinate destinationTranslation = getCoordinate(destinationLocation);
+  Angle      startingRotation       = DIRECTION_TO_ANGLE().value(m_startingDirection);
+
+  Coordinate currentTranslation = startingTranslation * (1.0 - fraction) + destinationTranslation * fraction;
+  Angle currentRotation         = startingRotation    * (1.0 - fraction) + destinationRotation    * fraction;
+
+  // Calculate the current translation and rotation for a curv turn
```

Comment:
This function is getting a bit too big. Can you create a separate helper function for the curve turn logic?